### DISCUSSION TITLE
How to use UnoCSS preflights using rules rather than raw CSS?

### DISCUSSION THREAD

**Discussion by raf202** [unocss/unocss#3093] [Q&A]
Using Uno on a project. I'm trying to do something like a @layer base on tailwind, that is, adding global CSS using rules. In Uno apparently the way to go is with preflights, described in this link: and uses this code:

```
preflights: [
  {
    getCSS: ({ theme }) => `
      * {
        color: ${theme.colors.gray?.[700] ?? '#333'};
        padding: 0;
        margin: 0;
      }
    `
  }
]
```
I want to load these preflights but using rules rather than raw CSS, something like this:

```
preflights: [
  {
    getCSS: ({ theme }) => `
      button {
      'margin-3 border-5 rounded-2px'
      }
    `
  }
]
```
Any ideas?

**Comment by snakemastr** [comment]
Better late than never I myself have solved it this way:
```
preflights: [
    {
      getCSS: ({ theme }) => `
              
        body {
          --uno: text-base font-body;
        }

     `
    }
  ],
```

**Reply by wtto00** [reply]
This `--uno` used `transformerDirectives` is not woring in postcss. Not sure if it works in Vite.

**Comment by LearningUnity** [comment]
'!margin-3 !border-5 !rounded-2px'

**Reply by wtto00** [reply]
I create a example project: https://codesandbox.io/p/github/rspack-contrib/rsbuild-codesandbox-example/csb-89wjsv/draft/clever-roman

```js
// uno.config.mjs
import { defineConfig, presetUno } from 'unocss';

export default defineConfig({
  content: {
    filesystem: ['./src/**/*.{html,js,ts,jsx,tsx,css}'],
  },
  presets: [presetUno()],
  preflights: [
    {
      getCSS: ({ theme }) => `
        body {
          color: blue;
          --uno: bg-red;
        }
      `
    }
  ]
});
```

```css
/* src/index.css */
@unocss preflights;
@unocss default;

@unocss;

body {
    --uno: font-italic;
    color: green;
}
```

- The output result of `font-size: 36px;` in `preflights` is correct, which is `font-size: 36px;`.
- The output result of `--uno: bg-red;` in `preflights` is not correct, which is `--uno: bg-red;` with no processing.
- The output result of `color: green;` in `index.css` is correct, which is `color: green;`.
- The output result of `--uno: font-italic` in `index.css` is not correct, which is `--uno: font-italic` with no processing.
